source/common/http/filter/lua/lua_filter.cc

```diff
@@ -116,8 +116,8 @@ int StreamHandleWrapper::luaRespond(lua_State* state) {
 
   uint64_t status;
   if (headers->Status() == nullptr ||
-      !StringUtil::atoul(headers->Status()->value().c_str(), status) || status < 200 ||
-      status >= 500) {
+      !StringUtil::atoul(headers->Status()->value().c_str(), status) || status < 100 ||
```

Comment:
Informational (1xx) status codes have special meaning and we should not allow a Lua script to send them, as they will confuse the connection manager, so that part was correct, but the error message below is wrong. But the other one was a bug. Thank you for fixing. We should have a test though. Do you mind changing one of the response tests to use a 5xx status code? If you don't want to bother I can fix this later today.

Reply:
I am leaving town this afternoon and will be off for a few days. Otherwise i would be able to fix it after i get back.

Reply:
Either way. If you would like to contribute we can wait, otherwise I'm happy to fix it. If you want to go that route just close the PR and I will do another PR later today.

Reply:
You can add the fix, i would rather come back to it merged than wait

Reply:
https://github.com/envoyproxy/envoy/pull/2034